src/toil/batchSystems/mesos/__init__.py

```diff
@@ -27,22 +31,110 @@
     'cores'))
 
 
-class ResourceRequirement( namedtuple('_ResourceRequirement', (
-    # Number of bytes (!) needed for a task
-    'memory',
-    # Number of CPU cores needed for a task
-    'cores',
-    # Number of bytes (!) needed for task on disk
-    'disk',
-    # True, if job can be run on a preemptable node, False otherwise
-    'preemptable'))):
+class JobQueue(object):
+
+    def __init__(self):
+        # mapping of jobTypes to queues of jobs of that type
+        self.queues = {}
+        # list of jobTypes in decreasing resource expense
+        self.sortedTypes = []
+
+    def insertJob(self, job, jobType):
+        # logn insertion
```

Comment:
Nitpick: I think this is O(n), because list insertion is worst-case O(n) (the elements after the insertion need to be shifted). That said, I think it's probably not worth using a tree or something to get O(log n), O(n) is most likely totally fine, because shifting the elements is insanely fast. I doubt the asymptotics really matter with less than millions of job types.